joint_publisher:
  type: "joint_state_controller/JointStateController"
  publish_rate: 50

manipulation_controller:
  type: manipulation_royalpanda/ManipulationController
  simulation: false
  fixed_base: $(arg fixed_base)

  arm_id: panda
  joint_names: [ "panda_joint1", "panda_joint2", "panda_joint3", "panda_joint4", "panda_joint5", "panda_joint6", "panda_joint7" ]
  i_gains: [10.0, 10.0, 10.0, 10.0, 10.0, 10.0, 10.0]  # this is actually an integral gain
  d_gains: [10.0, 10.0, 10.0, 10.0, 10.0, 10.0, 10.0]
  arm_I_max: 0.1

  state_topic: /observer/state
  nominal_state_topic: /x_nom
  base_twist_topic: /ridgeback_velocity_controller/cmd_vel

  policy_update_rate: 0.0  # execute as fast as possible
  reference_update_rate: 10.0
  publish_ros: true
  ros_publish_rate: 10.0
  static_optimization: false

  cost:
    regularization: 0.0
    linear_weight: 100.0
    linear_weight_opening: 100.0

    angular_weight: 10.0
    angular_weight_opening: 1.0

    min_dist: 0.0
    max_reach: 1.0
    reach_weight: 100
    reach_weight_slope: 10

    obstacle_weight: 0
    obstacle_weight_slope: 0
    obstacle_radius: 0.8

    contact_weight: 10.0

    object_weight: 10
    object_tolerance: 0.01

    grasp_translation_offset: [ 0.0, 0.0, 0.0 ]
    grasp_orientation_offset: [ 0, -0.7071068, 0, 0.7071068 ]    # -90deg around y

    upper_joint_limits: [ 2.0, 2.0, 6.28, 2.8973, 1.7628, 2.8973, 0.0698, 2.8973, 3.7525, 2.8973 ]
    lower_joint_limits: [ -2.0, -2.0, -6.28, -2.8973, -1.7628, -2.8973, -3.0718, -2.8973, -0.0175, -2.8973 ]

    joint_limit_cost: 1000 #1000
    joint_limit_slope: 1000000 #100

  object_reference_position: 1.57   

  base_gains:
    kp: [0.0, 0.0, 0.0]
    kd: [1000.0, 1000.0, 1000.0]

  arm_gains: 
    kp: [0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0]
    kd: [10.0, 10.0, 10.0, 10.0, 10.0, 10.0, 10.0]

  gripper_gains:
    kp: [100.0, 100.0]
    kd: [50.0, 50.0]

panda_position_controller:
  type: effort_controllers/JointGroupPositionController
  joints: ["panda_joint1",
           "panda_joint2",
           "panda_joint3",
           "panda_joint4",
           "panda_joint5",
           "panda_joint6",
           "panda_joint7"]
  panda_joint1:
    pid: { p: 120, d: 5, i: 0.0, i_clamp: 10000 }
  panda_joint2:
    pid: { p: 300, d: 10, i: 0.02, i_clamp: 10000 }
  panda_joint3:
    pid: { p: 180, d: 5, i: 0.01, i_clamp: 1 }
  panda_joint4:
    pid: { p: 180, d: 7, i: 0.01, i_clamp: 10000 }
  panda_joint5:
    pid: { p: 120, d: 7, i: 0.01, i_clamp: 1 }
  panda_joint6:
    pid: { p: 70, d: 5, i: 0.01, i_clamp: 1 }
  panda_joint7:
    pid: { p: 20, d: 2, i: 0.0, i_clamp: 1 }